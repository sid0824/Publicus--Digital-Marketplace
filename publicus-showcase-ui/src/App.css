.root {
  flex-grow: 1;
}

.title {
  flex-grow: 1;
}

Link {
  text-decoration: none;
  color: white;
}

html, body {
    max-width: 100%;
    overflow-x: hidden;
    scroll-behavior: smooth;
}
