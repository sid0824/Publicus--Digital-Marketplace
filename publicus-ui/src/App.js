import 'antd/dist/antd.css';
import React from 'react'
import {Layout, Menu, Icon, message, Button} from 'antd';
import {Link, withRouter} from "react-router-dom";
import RoutePaths from "./core/RoutePaths";
import Routes from "./core/Routes";
import {API, Auth} from "aws-amplify";
import { AuthContext } from './AuthContext';
import {ApiName, StripeKey} from "./core/Config";
import {Elements, StripeProvider} from 'react-stripe-elements';

const { Content, Sider } = Layout;

const content = {
  flexGrow: 1,
  minHeight: 360,
  overflow: 'auto',
};

const styles = {
  root: { flexGrow: 1, zIndex: 1, overflow: 'hidden', position: 'relative', display: 'flex', minHeight: '100vh' },
  titleContainer: { margin: '5px', whiteSpace: 'nowrap', textAlign: 'center', height: '50px', lineHeight: '50px', display: 'flex', flexDirection: 'row', justifyContent: 'center' },
  title: { marginLeft: 10 , color: 'white', fontSize: '15px', fontWeight: 'bold', marginRight: '14px' },
  sider: { height: '100vh', position: 'fixed', left: 0 },
  contentExpanded: { ...content, marginLeft: '200px', height: '100vh' },
  contentRetracted: { ...content, marginLeft: '80px', height: '100vh'  }
};

// Display titles for each site path
const titles = {
  [RoutePaths.Home]: 'Home',
  [RoutePaths.CreateAccount]: 'Create Account',
  [RoutePaths.CreateForm]: 'Create Form',
  [RoutePaths.Login]: 'Login',
  [RoutePaths.MyForms]: 'My Forms',
  [RoutePaths.Marketplace]: 'Marketplace',
  [RoutePaths.CompleteForm]: 'Complete Form',
  [RoutePaths.ViewResults]: 'Results',
  [RoutePaths.MyAccount]: 'My Account',
  [RoutePaths.Administration]: 'Administration',
};

/**
 * Provides navigation bar and authentication logic. Provides authentication values to all children
 * through context
 */
export class App extends React.Component {

  state = {
    adminUser: false,
    collapsed: false, // Is the navigation bar collapsed
    authenticated: false, // Is the current user authenticated
    loading: true,// Is the current user still authenticated,
    userId: ''
  };

  componentDidMount = async () => {
    try {
      this.setState({
        loading: true
      });
      await Auth.currentSession();
      await this.setUserHasAuthenticated(true);
      this.setState({
        loading: false
      });
    } catch (err) {
      console.error(err.message);
      this.setState({
        loading: false
      });
    }
  };

  /**
   * Set the collapsed value of the navigation bar
   * @param collapsed - new value
   */
  collapseSlider = (collapsed) => {
    this.setState({
      collapsed
    })
  };

  /**
   * Signout the current user
   */
  signOut = async () => {
    try {
      await Auth.signOut();
      this.setState({
        authenticated: false,
        adminUser: false
      })
    } catch (err) {
      message.error(err.message);
    }
  };

  /**
   * Set the new value of the user being authenticated
   * @param authenticated - new value
   */
  setUserHasAuthenticated = async (authenticated) => {
    let adminUser = false;
    let userId = '';
    let avatar = '';
    if (authenticated) {
      const userValues = await API.get(ApiName, "/getUserValues", {}); // change to get use values
      adminUser = !!userValues.admin;
      userId = userValues.userId;
      avatar = userValues.avatar;
    }
    this.setState({
      authenticated,
      adminUser,
      userId,
      avatar
    })
  };

  render() {
    const { collapsed, authenticated, loading, adminUser, userId, avatar } = this.state;
    const { location } = this.props;

    const authContext = {
      authenticated,
      setUserHasAuthenticated: this.setUserHasAuthenticated,
      adminUser,
      userId,
      avatar
    };

    const title = titles[location.pathname] || '';

    return (
      <StripeProvider apiKey={StripeKey}>
        <Elements>
          <Layout style={styles.root}>
            <Sider
              collapsed={collapsed}
              collapsible
              onCollapse={this.collapseSlider}
              style={styles.sider}
            >
              <div style={styles.titleContainer}>
                {
                  !collapsed && <div>
                    <Icon type="yuque" style={{ color: 'white', fontSize: 22, marginTop: 15, marginRight: 4 }} />
                  </div>
                }
                <div style={styles.title}>
                  Publicus
                </div>
              </div>
              <Menu
                theme="dark"
                defaultSelectedKeys={[RoutePaths.Home]}
                mode="inline"
                selectedKeys={[location.pathname]}
              >
                <Menu.Item key={RoutePaths.Home}>
                  <Link to={RoutePaths.Home}>
                    <Icon type="home" />
                    <span>
                      Home
                    </span>
                  </Link>
                </Menu.Item>
                <Menu.Item key={RoutePaths.CreateForm}>
                  <Link to={RoutePaths.CreateForm}>
                    <Icon type="form"/>
                    <span>Create Form</span>
                  </Link>
                </Menu.Item>
                <Menu.Item key={RoutePaths.Marketplace}>
                  <Link to={RoutePaths.Marketplace}>
                    <Icon type="pay-circle"/>
                    <span>Marketplace</span>
                  </Link>
                </Menu.Item>
                <Menu.Item key={RoutePaths.MyForms}>
                  <Link to={RoutePaths.MyForms}>
                    <Icon type="align-left"/>
                    <span>View My Forms</span>
                  </Link>
                </Menu.Item>
                  <Menu.Item key={RoutePaths.MyAccount}>
                    <Link to={RoutePaths.MyAccount}>
                      <Icon type="profile" />
                      <span>My Account</span>
                    </Link>
                  </Menu.Item>
                {
                  authenticated && adminUser &&
                    <Menu.Item key={RoutePaths.Administration}>
                      <Link to={RoutePaths.Administration}>
                        <Icon type="build" />
                        <span>Administration</span>
                      </Link>
                    </Menu.Item>
                }
              </Menu>
            </Sider>
            <Content style={collapsed ? styles.contentRetracted : styles.contentExpanded}>
              <div style={{
                backgroundColor: '#6fa6ed', height: '60px', display: 'flex', flexDirection: 'row', justifyContent: 'space-between',
              }}
              >
                <div style={{  height: '100%', marginLeft: '15px' }}>
                  <h1 style={{ lineHeight: '60px', fontSize: 30 }}>
                    {title}
                  </h1>
                </div>
                <div style={{  height: '100%', display: 'flex', justifyContent: 'flex-end', alignItems: 'center', marginRight: 10 }}>
                  {authenticated
                    ? <Button
                        onClick={this.signOut}
                        style={{ marginRight: 10 }}
                        type={"primary"}
                      >
                        Sign Out
                      </Button>
                    : <div>
                        <Link
                          to={RoutePaths.CreateAccount}
                          style={{ height: '100%', marginRight: 6 }}
                        >
                          <Button
                            style={{ marginRight: 0 }}
                            type={"primary"}
                          >
                            Create Account
                          </Button>
                        </Link>
                        <Link
                          to={RoutePaths.Login}
                          style={{ height: '100%', marginRight: 3 }}
                        >
                          <Button
                            style={{ marginRight: 0 }}
                            type={"primary"}
                          >
                            Login
                          </Button>
                      </Link>
                      </div>
                  }
                </div>
              </div>
              <div style={{ padding: '10px', height: 'calc(100% - 84px)' }}>
                <AuthContext.Provider value={authContext}>
                  {loading ?  <div style={{ width: '100%', height: '100%', display: 'flex', justifyContent: 'center', paddingTop: 200 }}>
                    <Icon type="loading" style={{ fontSize: 70, marginTop: 8 }} />
                  </div> : <Routes />}
                </AuthContext.Provider>
              </div>
            </Content>
          </Layout>
        </Elements>
      </StripeProvider>
    );
  }
}

export default withRouter(App);
