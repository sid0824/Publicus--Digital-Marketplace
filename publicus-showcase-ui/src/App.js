import React, { Component } from "react";
import './App.css';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import Button from '@material-ui/core/Button';
import { Link } from 'react-router-dom';
import Routes from "./routes";


class App extends Component  {

  render() {
    return (
      <div className={'root'}>
        <AppBar position="relative">
          <Toolbar>
            <Typography variant="h6" className={'title'} component={Link} to={'/home'}
                        style={{textDecoration: 'none', color: 'white'}}>
              Publicus
            </Typography>
            <Button color="inherit" component={Link} to={'/home/preregister'}>Pre-register</Button>

          </Toolbar>
        </AppBar>
        <Routes/>
      </div>
    );
  }
}


export default App;
