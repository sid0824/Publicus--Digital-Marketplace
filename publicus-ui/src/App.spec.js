import toJson from 'enzyme-to-json';
import React from "react";
import {shallow, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {App} from "./App";
import {API, Auth} from "aws-amplify";

configure({ adapter: new Adapter() });

const userValues = {
  "userId": "ap-southeast-2:91ecfa27-a7c0-45a8-9f45-c93885409ca7",
  "email": "publicusnoreply@gmail.com",
  "admin": 1,
  "avatar": "https://publicus-file-uploads.s3.amazonaws.com/avatar/9b82dbf0-f47b-11e9-a70a-e16b9b703746",
  "firstname": "Admin",
  "lastname": "Account"
};

describe('App', () => {

  let wrapper;
  beforeEach(() => {
    Auth.currentSession = jest.fn().mockResolvedValue(null);
    Auth.signOut = jest.fn().mockResolvedValue(null);
    API.get = jest.fn().mockResolvedValue(userValues);
    wrapper = shallow(
      <App
        location={{
          pathName: 'myForms'
        }}
      />
    );
  });

  test('should render authorised admin route', (done) => {
    setTimeout(() => {
      expect(toJson(wrapper)).toMatchSnapshot();
      done();
    })
  });


  test('should log out of the site', (done) => {
    setTimeout(async () => {
      await wrapper.instance().signOut();
      expect(wrapper.state()).toMatchSnapshot();
      done();
    })
  });

});