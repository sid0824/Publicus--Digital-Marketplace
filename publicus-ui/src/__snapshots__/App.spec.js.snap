// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should log out of the site 1`] = `
Object {
  "adminUser": false,
  "authenticated": false,
  "avatar": "https://publicus-file-uploads.s3.amazonaws.com/avatar/9b82dbf0-f47b-11e9-a70a-e16b9b703746",
  "collapsed": false,
  "loading": false,
  "userId": "ap-southeast-2:91ecfa27-a7c0-45a8-9f45-c93885409ca7",
}
`;

exports[`App should render authorised admin route 1`] = `
<Provider
  apiKey="pk_test_qBUWBnakwQI3TCy8jqDOGdyS00gXGU5q9k"
>
  <Elements>
    <Adapter
      style={
        Object {
          "display": "flex",
          "flexGrow": 1,
          "minHeight": "100vh",
          "overflow": "hidden",
          "position": "relative",
          "zIndex": 1,
        }
      }
    >
      <Sider
        collapsed={false}
        collapsible={true}
        onCollapse={[Function]}
        style={
          Object {
            "height": "100vh",
            "left": 0,
            "position": "fixed",
          }
        }
      >
        <div
          style={
            Object {
              "display": "flex",
              "flexDirection": "row",
              "height": "50px",
              "justifyContent": "center",
              "lineHeight": "50px",
              "margin": "5px",
              "textAlign": "center",
              "whiteSpace": "nowrap",
            }
          }
        >
          <div>
            <Icon
              style={
                Object {
                  "color": "white",
                  "fontSize": 22,
                  "marginRight": 4,
                  "marginTop": 15,
                }
              }
              type="yuque"
            />
          </div>
          <div
            style={
              Object {
                "color": "white",
                "fontSize": "15px",
                "fontWeight": "bold",
                "marginLeft": 10,
                "marginRight": "14px",
              }
            }
          >
            Publicus
          </div>
        </div>
        <Menu
          defaultSelectedKeys={
            Array [
              "/",
            ]
          }
          mode="inline"
          selectedKeys={
            Array [
              undefined,
            ]
          }
          theme="dark"
        >
          <MenuItem
            key="/"
          >
            <Link
              to="/"
            >
              <Icon
                type="home"
              />
              <span>
                Home
              </span>
            </Link>
          </MenuItem>
          <MenuItem
            key="/createForm"
          >
            <Link
              to="/createForm"
            >
              <Icon
                type="form"
              />
              <span>
                Create Form
              </span>
            </Link>
          </MenuItem>
          <MenuItem
            key="/marketplace"
          >
            <Link
              to="/marketplace"
            >
              <Icon
                type="pay-circle"
              />
              <span>
                Marketplace
              </span>
            </Link>
          </MenuItem>
          <MenuItem
            key="/myForms"
          >
            <Link
              to="/myForms"
            >
              <Icon
                type="align-left"
              />
              <span>
                View My Forms
              </span>
            </Link>
          </MenuItem>
          <MenuItem
            key="/myAccount"
          >
            <Link
              to="/myAccount"
            >
              <Icon
                type="profile"
              />
              <span>
                My Account
              </span>
            </Link>
          </MenuItem>
          <MenuItem
            key="/administration"
          >
            <Link
              to="/administration"
            >
              <Icon
                type="build"
              />
              <span>
                Administration
              </span>
            </Link>
          </MenuItem>
        </Menu>
      </Sider>
      <Adapter
        style={
          Object {
            "flexGrow": 1,
            "height": "100vh",
            "marginLeft": "200px",
            "minHeight": 360,
            "overflow": "auto",
          }
        }
      >
        <div
          style={
            Object {
              "backgroundColor": "#6fa6ed",
              "display": "flex",
              "flexDirection": "row",
              "height": "60px",
              "justifyContent": "space-between",
            }
          }
        >
          <div
            style={
              Object {
                "height": "100%",
                "marginLeft": "15px",
              }
            }
          >
            <h1
              style={
                Object {
                  "fontSize": 30,
                  "lineHeight": "60px",
                }
              }
            />
          </div>
          <div
            style={
              Object {
                "alignItems": "center",
                "display": "flex",
                "height": "100%",
                "justifyContent": "flex-end",
                "marginRight": 10,
              }
            }
          >
            <Button
              block={false}
              ghost={false}
              htmlType="button"
              loading={false}
              onClick={[Function]}
              style={
                Object {
                  "marginRight": 10,
                }
              }
              type="primary"
            >
              Sign Out
            </Button>
          </div>
        </div>
        <div
          style={
            Object {
              "height": "calc(100% - 84px)",
              "padding": "10px",
            }
          }
        >
          <ContextProvider
            value={
              Object {
                "adminUser": true,
                "authenticated": true,
                "avatar": "https://publicus-file-uploads.s3.amazonaws.com/avatar/9b82dbf0-f47b-11e9-a70a-e16b9b703746",
                "setUserHasAuthenticated": [Function],
                "userId": "ap-southeast-2:91ecfa27-a7c0-45a8-9f45-c93885409ca7",
              }
            }
          >
            <_default />
          </ContextProvider>
        </div>
      </Adapter>
    </Adapter>
  </Elements>
</Provider>
`;
